content/docs/start/model-management/model-cicd.md

```diff
@@ -0,0 +1,201 @@
+which deploys a specific version of the model to an Amazon Sagemaker endpoint
+for inference whenever it is assigned to a stage.
+
+</admon>
+
+Go to the `.github/workflows/deploy-model-template.yml`. This is the file that
+GitHub uses to run our CICD workflow. You can see
+[runs of this workflow](https://github.com/iterative/example-get-started-experiments/actions/workflows/deploy-model-template.yml)
+in our example repository.
+
+At the beginning of the workflow file you will see this code
+
+```yaml
+on:
+  # the workflow is triggered whenever a tag is pushed to the repository
+  push:
+    tags:
+      - '*'
+```
+
+The code tells GitHub to run the workflow every time a tag is pushed to the
+repository.
+
+This means that the workflow will run whenever we run model registry actions,
+but we also want it to limit to specific ones for our specific workflow. That's
+where our GTO GitHub action comes into play - in the "parse" job of our workflow
+it parses all tags and if they are GTO tags, it gives us the name of the model,
+its version, stage (if any) and the event in the model registry.
+
+This is captured in the "parse" job which you can simply copy and paste into
+most CICD jobs of your own.
+
+```yaml
+# This job parses the git tag with the GTO GitHub Action to identify model registry actions
+parse:
+  runs-on: ubuntu-latest
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: 'Parse GTO tag'
+      id: gto
+      uses: iterative/gto-action@v2
+  outputs:
+    event: ${{ steps.gto.outputs.event }}
+    name: ${{ steps.gto.outputs.name }}
+    stage: ${{ steps.gto.outputs.stage }}
+    version: ${{ steps.gto.outputs.version }}
+```
+
+<admon type="tip">
+
+If you cannot or don't want to use the GTO GitHub Action you can also use GTO
+directly with the [gto check-ref](/doc/gto/command-reference/check-ref) command.
+
+</admon>
+
+The next job called "deploy-model" actually performs the action. First, it uses
+the outputs of the parse job and checks whether the action should be performed.
+If the tag was produced by the model registry and if the corresponding action
+was assignment to the "prod" stage, it proceeds with the rest of the workflow.
+
+```yaml
+deploy-model:
+  needs: parse
+  if:
+    "${{ needs.parse.outputs.event == 'assignment' }} && ${{
+    needs.parse.outputs.stage == 'prod' }}"
+```
+
+The next two steps step of the workflow checkout the Git repository and set up
+DVC (using a GitHub Action, but this can also be done manually, for example with
+pip).
+
+This allows us to run `dvc artifacts get` in the last step of the workflow to
+download the correct version of the model which can then be deployed or
+otherwise used in our CICD.
+
+```yaml
+- name: Get Model For Deployment
+  run: |
+    dvc artifacts get ${{ github.repositoryUrl }} ${{ needs.parse.outputs.name }} --rev ${{ needs.parse.outputs.version }}
```

Comment:
```suggestion
    dvc artifacts get . ${{ needs.parse.outputs.name }} --rev ${{ needs.parse.outputs.version }}
```

Do we need the repo url here? If we can drop it, it also simplifies the explanation below.

Reply:
I guess I have to test this but wouldn't we need the checkout action with your suggestion?  Or maybe I am a bit confused about how `dvc artifacts get` works.

Reply:
Oh, so the checkout was related to the parse job, I missed that. However, I think we can probably do away without the checkout action in the deploy job if we keep the GitHub reference.

I am not sure what is simpler, but I kind of prefer using the GH reference it is less code and you don't really need the checkout action. Again, I will try this out.

Reply:
No strong opinion on which way you go here

Reply:
We definitely need the Studio token here though...or remote credentials. I will go with the token as we do above.